### DISCUSSION TITLE
Probably a silly question but…what exactly does the Overwrite setting do?

### DISCUSSION THREAD

**Discussion by ThisByzantineConduit** [upscayl/upscayl#683] [Q&A] [answered]
What exactly does the Overwrite Previous Upscale option do? The way I interpreted the language is that by default images that have already been upscaled wont be processed and upscaled again, and that the overwrite option enables it to be processed again.

However when I, for example, bulk upscale a folder that partially contains previously upscaled images, they previously upscaled ones are still upscaled and processed again despite overwriting being disabled. I bet I'm just being silly and misunderstanding something, so please clarify 😅.

**Comment by aaronliu0130** [comment]
It currently doesn't work for batch Upscayl.

**Reply by ThisByzantineConduit** [reply]
Ah ok, but how does it work on a single photo? Like if I take a preciously upscaled photo and run it again the output resolution stays the same without overwrite enabled?

Thanks for the reply!

**Reply by aaronliu0130** [reply, marked as answer]
On a single photo, if Upscayl realizes that a file processed by Upscayl with the same scale and model exists in the output folder, it will not Upscayl again, "instantly finish", and display our diff view of the before/after.
